yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -123,6 +123,7 @@ private[spark] class Client(
     appContext.setQueue(args.amQueue)
     appContext.setAMContainerSpec(containerContext)
     appContext.setApplicationType("SPARK")
+    setNodeLabelExpressions(appContext)
```

Comment:
I thought I had checked this, but it looks like in fact it will.  Reverting to the old patch.